### DISCUSSION TITLE
Add And Remove Items when Collection is Changed

### DISCUSSION THREAD

**Discussion by bbaileyHOM** [gismofx/toast_ui.blazor_calendar#31] [Ideas]
First, thank you for doing the leg work on making this work in Blazor!

Is there a reason that the items on the calendar aren't updated with the Schedules collection is changed?  Is there another way in which changes to the calendar are expected to be add/removed from the calendar?  I wasn't able to determine if there was.

I downloaded the project and the made the following changes to TUISchedule.cs:

added:
`        private ICollection<TUISchedule> _schedules;`

altered:
```
        [Parameter]
        public ICollection<TUISchedule> Schedules
        {
            get => _schedules;
            set
            {
                _schedules = value;
                
                if (_schedules == null || CalendarInterop == null) return;

                ClearCalendar();
                CalendarInterop.CreateSchedulesAsync(_schedules);

                InvokeAsync(StateHasChanged);
            }
        }
```

Any time the collection changes the calendar is cleared and all schedules are recreated.  It might be better to use ObservableCollection<T> and add/remove/update but that is a bit more complicated and I am not sure if there is a lot of  value in it.

**Comment by gismofx** [comment]
Thanks, I'm happy to contribute the community. And thanks for the idea and thoughts! 

Updates can come from the UI and those changes are synced with collection. 

Correct me if I misunderstood, I think you're wanting to make changes or add events from code/server-side/C# and push just those incremental schedules down to the calendar? 

Happy to collaborate, just want to make sure I'm in the same page.

**Reply by bbaileyHOM** [reply]
Yes, I'm essentially posting items to the calendar that I have in a database.  And in my case they aren't points in time but schedules with occurrence rules so I'm not even using the built in functionality for creating the items.

**Reply by gismofx** [reply]
I believe TUI supports recurrence rules. Have you tried calling `CreateSchedulesAsync` method in the interop? I wonder if that will help you to add schedules from code.

**Reply by bbaileyHOM** [reply]
I had thought that I had tried using that method and it wasn't working for me for some reason, but I just tried it again and that works as well.

From what I have read TUI doesn't have built in recurrence functionality but provides a property for implementing if you like.  And that wouldn't probably work well for me anyway.

Thanks again!
